File: ProyectoCsharp/Modelo/Merma.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


public class Merma : Articulo
{
    private TTipoMerma motivo_de_baja;

    public Merma()
    {
    }

    public Merma(long id_articulo, string codigo, string descripcion, double cantidad, TUnidad unidad, DateTime fecha_caducidad, Categoria categoria, TTipoMerma motivo_de_baja, double precio) 
         : base(id_articulo, codigo, descripcion, cantidad, unidad, fecha_caducidad, categoria, precio)
    {
        Motivo_de_baja = motivo_de_baja;
    }

    public TTipoMerma Motivo_de_baja { get => motivo_de_baja; set => motivo_de_baja = value; }
}

File: TrabajoJavaLp2/src/modelo/Merma.java
package modelo;

import java.util.Date;

public class Merma  extends Articulo
{
    private String motivo_de_baja;

    public Merma()
    {
        motivo_de_baja = null;
    }

    public Merma(long id_articulo, String codigo, String descripcion, double cantidad,double precio, TUnidad unidad, Date fecha_caducidad, Categoria categoria, String motivo_de_baja) 
         
    {
	super(id_articulo, codigo, descripcion, cantidad, precio, unidad, fecha_caducidad, categoria);
        motivo_de_baja = motivo_de_baja;
    }

	public void setMotivo_de_baja(String motivo){
		this.motivo_de_baja = motivo;
	}
	
	public String getMotivo_de_baja(){
		return this.motivo_de_baja;
	}
}

